### ISSUE TITLE
Adding DbFields with standard values to DbTemplates

### ISSUE THREAD

**Issue by apoetzsch** [sergeyshushlyapin/Sitecore.FakeDb#129]
Hey there,
recently I tried to define a checkbox field on a DbTemplate and assign a standard value to this field. I used the following code to create the template with its field:
`new DbTemplate("Page", pageTemplateId) { new DbField("Hide") { Type = "checkbox" } }`
The problem is, I didn't found a way to define a standard value for that specific field. It seems that it's only possible to define text-based fields with standard values using `new DbTemplate("Page", pageTemplateId){{'myField', 'this is my standard value'}}`.

Having a look to the DbTemplate class I found the following method that does exactly what I was looking for: `DbTemplate.Add(DbField field, string standardValue)` but this method is protected? Is there a reason why this method is protected or would it be possible to change it to public.

Thanks for your help.
Axel

**Comment by sergeyshushlyapin** [comment]
I agree, it should be public. It's also important to _clone_ the field properly when adding to the standard values. Now only name and id properties are copied (you need Type as well.

**Comment by sergeyshushlyapin** [comment]
That is how it may look:
```
var templateId = ID.NewID;
const string checkboxStandardValue = "1";
using (var db = new Db
  {
    new DbTemplate("Page", templateId)
      {
        { new DbField("Hide") { Type = "checkbox" }, checkboxStandardValue }
      },
    new DbItem("home", ID.NewID, templateId)
  })
{
  var home = db.GetItem("/sitecore/content/home");

  Assert.True(new CheckboxField(home.Fields["Hide"]).Checked);
}
```

**Comment by sergeyshushlyapin** [comment]
Released as [v1.1.0](https://www.nuget.org/packages/Sitecore.FakeDb/). Thanks @apoetzsch for highlighting this question!

**Comment by apoetzsch** [comment]
Amazing - this helps a lot! Thank you very much.
